### PR TITLE
unassigned filters implemented. Filter By pill bug fixed

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Dual Merge: [BE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/466)
[Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/148)

Filter By pill bug also fixed

So many edge cases =-[

### CODE REVIEW

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx [hunk 1]
```diff
@@ -62,20 +62,16 @@ class BidderPortfolio extends Component {
   // the instance.
   onQueryParamUpdate = q => {
     const { query, page } = this.state;
-    let removePageandLimit = false;
-    if (isEqual(q, { limit: 'all' })) {
-      removePageandLimit = true;
-    }
```
Comment (elizabeth-jimenez) [inline src/Containers/BidderPortfolio/BidderPortfolio.jsx:68]: removed this code as it was no longer being used. Was for the Display Clients 'All' selection

File: src/actions/bidderPortfolio.js [hunk 2]
```diff
@@ -219,10 +234,23 @@ export function bidderPortfolioFetchData(query = {}) {
...
+        if (unassigned.includes('noBids')) {
+          query$.noBids = true;
+        }
+      }
     }
     if (!query$.ordering) {
       query$.ordering = BID_PORTFOLIO_SORTS.defaultSort;
     }
+    // eslint-disable-next-line no-console
+    console.log(query$);
```
Comment (mjoyce91) [inline src/actions/bidderPortfolio.js:253]: still needed?

**Comment by mjoyce91** [comment]
The screen width responsiveness of this is 💯 

Screen Shot 2021-01-12 at 12 50 55 PM

File: src/actions/bidderPortfolio.js (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/actions/bidderPortfolio.js:253]: for your convenience lol

**Comment by mjoyce91** [comment]
Can you create a new feature flag, let's say `"unassigned_bidders"`, so that we can hide the "Available bidders" dropdown option, as well as the "Unassigned Bidders" filter?

**Comment by elizabeth-jimenez** [comment]
> Can you create a new feature flag, let's say `"unassigned_bidders"`, so that we can hide the "Available bidders" dropdown option, as well as the "Unassigned Bidders" filter?

yep! good idea 😓  so many edge cases here

**Comment by mjoyce91** [comment]
It appears that the Unassigned Bidders filters are persisting and being passed to the request, even after a Filter By option is selected that disables the Unassigned Bidders dropdown

Screen Shot 2021-01-21 at 10 05 34 AM


Screen Shot 2021-01-21 at 10 05 40 AM

**Comment by elizabeth-jimenez** [comment]
> It appears that the Unassigned Bidders filters are persisting and being passed to the request, even after a Filter By option is selected that disables the Unassigned Bidders dropdown
> 
> Screen Shot 2021-01-21 at 10 05 34 AM
> 
> Screen Shot 2021-01-21 at 10 05 40 AM

@mjoyce91 Should be fixed!

File: src/actions/bidderPortfolio.js [hunk 3]
```diff
@@ -236,13 +236,14 @@ export function bidderPortfolioFetchData(query = {}) {
     }
     if (get(query, 'hasHandshake') === 'available_bidders') {
       query$ = omit(query$, ['hasHandshake']);
-      if (includes(unassigned, 'noHandshake')) {
+      const UAvalues = flatMap(unassigned, a => a.value);
```
Comment (mjoyce91) [inline src/actions/bidderPortfolio.js:239]: couldn't you just use regular `map` here?

File: src/Components/BidderPortfolio/BidControls/BidControls.jsx [hunk 4]
```diff
@@ -32,6 +45,19 @@ class BidControls extends Component {
...
       this.setState({ proxyCdos: this.props.selection }, this.generatePills);
     }
+    const BID_PORTFOLIO_FILTERS$ = BID_PORTFOLIO_FILTERS;
+    if (!useUnassignedFilter()) {
+      BID_PORTFOLIO_FILTERS$.options = BID_PORTFOLIO_FILTERS.options.filter(b => b.value !== 'available_bidders');
+    }
+    this.setState({ filterBy: BID_PORTFOLIO_FILTERS$.options[
+      findIndex(BID_PORTFOLIO_FILTERS$.options, (o) => o.value ===
+        this.props.defaultHandshake)] });
+    if (this.props.defaultHandshake === 'available_Bidders' && this.state.bidSeasons.length) {
```
Comment (mjoyce91) [inline src/Components/BidderPortfolio/BidControls/BidControls.jsx:55]: whats up with the case of `available_Bidders`? How about `available_bidders` or `availableBidders`?

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidControls.jsx:55]: woow, never even saw that. Good catch!

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx [hunk 5]
```diff
@@ -62,39 +62,23 @@ class BidderPortfolio extends Component {
...
-      this.getBidderPortfolio(removePageandLimit);
+      this.getBidderPortfolio();
     });
   };
 
   // Form our query and then retrieve bidders.
-  getBidderPortfolio(removePageandLimit) {
-    const query = this.createSearchQuery(removePageandLimit);
+  getBidderPortfolio() {
+    const query = this.createSearchQuery();
```
Comment (mjoyce91) [inline src/Containers/BidderPortfolio/BidderPortfolio.jsx:81]: Why was `removePageandLimit` removed?

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx [hunk 6]
```diff
@@ -186,7 +162,7 @@ BidderPortfolio.propTypes = {
   fetchBidderPortfolioCounts: PropTypes.func.isRequired,
   fetchBidderPortfolioCDOs: PropTypes.func.isRequired,
   cdos: PropTypes.arrayOf(PropTypes.shape({})),
-  selectedSeasons: PropTypes.arrayOf(PropTypes.string), // eslint-disable-line
+  selectedSeasons: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])), // eslint-disable-line
```
Comment (mjoyce91) [inline src/Containers/BidderPortfolio/BidderPortfolio.jsx:165]: eslint-disable still needed?

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx [hunk 7]
```diff
@@ -195,6 +171,7 @@ BidderPortfolio.propTypes = {
   defaultHandshakeFilter: PropTypes.string,
   defaultSort: PropTypes.string,
   fetchAvailableBidders: PropTypes.func.isRequired,
+  selectedUnassigned: PropTypes.arrayOf(PropTypes.shape({})), // eslint-disable-line
```
Comment (mjoyce91) [inline src/Containers/BidderPortfolio/BidderPortfolio.jsx:174]: eslint-disable still needed?

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx (hunk 5 above)
Comment (elizabeth-jimenez) [reply src/Containers/BidderPortfolio/BidderPortfolio.jsx:81]: it was deprecated; it was from when we had the 'All' selection in the Display Clients dropdown

File: src/actions/bidderPortfolio.js (hunk 3 above)
Comment (elizabeth-jimenez) [reply src/actions/bidderPortfolio.js:239]: Where would be the fun in that 😄

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx (hunk 6 above)
Comment (elizabeth-jimenez) [reply src/Containers/BidderPortfolio/BidderPortfolio.jsx:165]: yea :( and the one on line 174

File: src/Containers/BidderPortfolio/BidderPortfolio.jsx (hunk 7 above)
Comment (elizabeth-jimenez) [reply src/Containers/BidderPortfolio/BidderPortfolio.jsx:174]: yea :/

File: src/Components/BidderPortfolio/BidControls/BidControls.jsx [hunk 8]
```diff
@@ -15,6 +18,14 @@ import CDOAutoSuggest from '../CDOAutoSuggest';
 import ResetFilters from '../../ResetFilters/ResetFilters';
 
 const useCDOSeasonFilter = () => checkFlag('flags.cdo_season_filter');
+const useUnassignedFilter = () => checkFlag('flags.unassigned_bidders');
+
+export function renderList({ items, selected, ...rest }) {
+  const getIsSelected = item => !!selected.find(f => f.value === item.value);
+  return items.map(item => (
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/BidControls/BidControls.jsx:25]: happy @mjoyce91 😭

**Review by mjoyce91 — APPROVED** [review]

**Pull request merged**